src/runtime_src/driver/zynq/drm/zocl/sched_exec.c

```diff
@@ -1621,6 +1628,41 @@ cq_check(void *data)
 	return 0;
 }
 
+static irqreturn_t sched_exec_isr(int irq, void *arg)
+{
+	struct drm_zocl_dev *zdev = arg;
+	void __iomem *regs;
+	int cu_idx;
+	volatile u32 __iomem *virt_addr;
+
+	SCHED_DEBUG("-> sched_exe_isr irq %d", irq);
+	for (cu_idx = 0; cu_idx < zdev->cu_num; cu_idx++) {
+		if (zdev->irq[cu_idx] == irq)
+			break;
+	}
+
+	SCHED_DEBUG("cu_idx %d interrupt handle", cu_idx);
+	if (cu_idx >= zdev->cu_num) {
+		/* This should never happen */
+		DRM_ERROR("Unknown isr irq %d, polling %d\n",
+			  irq, zdev->exec->polling_mode);
+		return IRQ_HANDLED;
+	}
+
+	virt_addr = zdev->regs + cu_idx_to_offset(zdev->ddev, cu_idx);
+	/* Clear all CU interrupts, will need to handle ap_done and ap_ready
+	 * interrupt after support dataflow exectution model
+	 */
+	virt_addr[3] = virt_addr[3];
```

Comment:
This looks weird from software point of view. We'd better add some comments on why we are doing this.

Reply:
Add comments